balancer_conn_wrappers.go

```diff
@@ -163,6 +163,22 @@ func (ccb *ccBalancerWrapper) RemoveSubConn(sc balancer.SubConn) {
 	ccb.cc.removeAddrConn(acbw.getAddrConn(), errConnDrain)
 }
 
+func (ccb *ccBalancerWrapper) UpdateAddresses(sc balancer.SubConn, addrs []resolver.Address) {
+	acbw, ok := sc.(*acBalancerWrapper)
+	if !ok {
+		return
+	}
+
+	ccb.mu.Lock()
+	if ccb.subConns == nil {
+		ccb.mu.Unlock()
+		return
+	}
+	ccb.mu.Unlock()
```

Comment:
I think I asked about this before and you didn't know (?) -- why is this check here?  It seems superfluous given the LB policy would have been calling UpdateAddresses directly previously.

Reply:
After looking at the code a little bit more, I realized that this check is in all methods mainly to make sure that we do not process any of these calls once the balancer_wrapper is closed (which is done when the ClientConn is closed, or the LB policy is switched out). That is the reason why I let this stay. Let me know what you think.

Reply:
Do we have similar checks in `acbw.UpdateAddresses`?  If so, I'd omit this to keep it as minimal as possible.  If not...why not?

Reply:
I think this is what is happening. `addrConn` is torn down when either the `ClientConn` is closed, or the balancer is switched out. And correspondingly `addrConn` has a state associated with itself.

Now, the existing methods of `acBalancerWrapper`, namely `UpdateAddresses()` and `Connect()` call into appropriate methods on the `addrConn` which eventually check the state of the `addrConn`, and if it is in `Shutdown` state, they end up doing nothing. 

So, we should be safe to omit this check here and rely on the checks underneath. What do you think?